addons/sale/models/sale.py

```diff
@@ -116,6 +116,9 @@ def _get_payment_type(self):
         self.ensure_one()
         return 'form'
 
+    def _search_invoice_ids(self, operator, value):
+        return ['&', '&', ('order_line.invoice_lines.move_id.type', 'in', ('out_invoice', 'out_refund')), ('id', 'in', self.ids), ('order_line.invoice_lines.move_id', operator, value)]
```

Comment:
puzzled about `('id', 'in', self.ids)` in this domain; this assumes all searches are made on a recordset. Won't this return *nothing* if done on the model itself?